src/hifz/dataserver.py

```diff
@@ -29,7 +29,7 @@ def read_entries(self, file_path: Path) -> list[Card]:
 class FileInputReader:
     """Dispatches the appropriate reader strategy based on file extension."""
 
-    def __init__(self):
+    def __init__(self) -> None:
```

Comment:
It may be worth enabling [ANN](https://docs.astral.sh/ruff/rules/missing-return-type-special-method/) for ruff.

Reply:
Good idea. Updated the pr to include ANN204 in the ruff configs.